core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/collections/javautilcollection/JavaCollectionFacet.java

```diff
@@ -24,28 +24,39 @@
 
 import com.google.common.collect.Collections2;
 
+import org.apache.isis.commons.internal.base._NullSafe;
 import org.apache.isis.core.metamodel.adapter.ObjectAdapter;
-import org.apache.isis.core.metamodel.adapter.mgr.AdapterManager;
+import org.apache.isis.core.metamodel.adapter.ObjectAdapterProvider;
 import org.apache.isis.core.metamodel.facetapi.FacetHolder;
 import org.apache.isis.core.metamodel.facets.collections.CollectionFacetAbstract;
 
 public class JavaCollectionFacet extends CollectionFacetAbstract {
 
-    private final AdapterManager adapterManager;
+    private final ObjectAdapterProvider adapterProvider;
 
-    public JavaCollectionFacet(final FacetHolder holder, final AdapterManager adapterManager) {
+    public JavaCollectionFacet(final FacetHolder holder, final ObjectAdapterProvider adapterProvider) {
         super(holder);
-        this.adapterManager = adapterManager;
+        this.adapterProvider = adapterProvider;
     }
-
+    
     @Override
-    public Collection<ObjectAdapter> collection(final ObjectAdapter wrappedCollection) {
-        final Collection<?> collectionOfUnderlying = collectionOfUnderlying(wrappedCollection);
-
+    public void init(final ObjectAdapter collection, final ObjectAdapter[] initData) {
+        final Collection<? super Object> pojoCollection = pojoCollection(collection);
+        pojoCollection.clear();
+        for (final ObjectAdapter element : initData) {
+            final Object pojo = element.getObject();
+            pojoCollection.add(pojo);
+        }
+    }
+    
+    @Override
+    public Collection<ObjectAdapter> collection(final ObjectAdapter collectionAdapter) {
+        final Collection<?> pojoCollection = pojoCollection(collectionAdapter);
+        
         //TODO [ahuber] java doc states, this is a live view, don't know if this is needed, 
         // or if a copy is sufficient
-        return Collections2.transform(collectionOfUnderlying,
-                ObjectAdapter.Functions.adapter_ForUsing(getAdapterManager()));
+        return Collections2.transform(pojoCollection,
```

Comment:
no change in behavior, but ...
Is this live-view actually required?

Reply:
It's probably would be fine to be a copy, but is there any need to change the behaviour?

Reply:
no need to change behavior yet; just a standing question, that prevents removal of guava's class Collections2